game-core/src/main/java/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -165,129 +163,100 @@ private void createAndSaveStats(final boolean showPhaseStats) {
+          text.append(p.getName()).append(',');
         }
       } else {
         text.append("none yet; game not over,");
       }
-      text.append("\n");
-      text.append("\n");
-      text.append("Resource Chart: ,");
-      text.append("\n");
+      text.append('\n');
+      text.append('\n');
+      text.append("Resource Chart: ,").append('\n');
       for (final Resource resource : gameData.getResourceList().getResources()) {
-        text.append(resource.getName()).append(",");
-        text.append("\n");
+        text.append(resource.getName()).append(',').append('\n');
       }
       // if short, we won't both showing production and unit info
       if (showPhaseStats) {
-        text.append("\n");
-        text.append("Production Rules: ,");
-        text.append("\n");
+        text.append('\n');
+        text.append("Production Rules: ,").append('\n');
         text.append("Name,Result,Quantity,Cost,Resource,\n");
         final Collection<ProductionRule> purchaseOptions = gameData.getProductionRuleList().getProductionRules();
         for (final ProductionRule pr : purchaseOptions) {
-          String costString = pr.toStringCosts().replaceAll("; ", ",");
-          costString = costString.replaceAll(" ", ",");
-          text.append(pr.getName()).append(",").append(pr.getResults().keySet().iterator().next().getName()).append(",")
-              .append(pr.getResults().getInt(pr.getResults().keySet().iterator().next())).append(",").append(costString)
-              .append(",");
-          text.append("\n");
+          final String costString = pr.toStringCosts().replaceAll(";? ", ",");
+          text.append(pr.getName()).append(',')
+              .append(pr.getResults().keySet().iterator().next().getName()).append(',')
+              .append(pr.getResults().getInt(pr.getResults().keySet().iterator().next())).append(',')
+              .append(costString).append(',')
+              .append('\n');
         }
-        text.append("\n");
-        text.append("Unit Types: ,");
-        text.append("\n");
+        text.append('\n');
+        text.append("Unit Types: ,").append('\n');
         text.append("Name,Listed Abilities\n");
         for (final UnitType unitType : gameData.getUnitTypeList()) {
           final UnitAttachment ua = UnitAttachment.get(unitType);
           if (ua == null) {
             continue;
           }
-          String toModify = ua.allUnitStatsForExporter();
-          toModify = toModify.replaceFirst("UnitType called ", "").replaceFirst(" with:", "")
-              .replaceAll("games.strategy.engine.data.", "").replaceAll("\n", ";").replaceAll(",", ";");
-          toModify = toModify.replaceAll("  ", ",");
-          toModify = toModify.replaceAll(", ", ",").replaceAll(" ,", ",");
-          text.append(toModify);
-          text.append("\n");
+          final String toModify = ua.allUnitStatsForExporter()
+              .replaceAll("UnitType called | with:|games.strategy.engine.data.", "")
```

Comment:
I know this error was in the old code, but shouldn't the dots be escaped?  That is, `games\\.strategy\\.engine\\.data\\.`.

Reply:
Indeed